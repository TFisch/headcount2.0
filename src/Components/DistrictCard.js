import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../styling/DistrictCard.css';

class DistrictCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: false
    };
  }

  handleClick = () => {
    if (this.state.selected) {
      let selected = this.state.selected;
      selected = !selected;
      this.setState({ selected });
      this.props.removeCompareCard(this.props.location);
    }

    if (this.props.comparedCards.length < 2) {
      let selected = this.state.selected;
      selected = !selected;
      this.setState({ selected });
      this.props.retrieveCompare(this.props);
      if (this.state.selected) {
        selected = !selected;
        this.props.removeCompareCard(this.props.location);
      }
    }
  };

  render() {
    return (
      <div
        className={
          !this.state.selected ? 'district-card' : 'district-card selected'
        }
        onClick={this.handleClick}
      >
        <h3 className="location-header">{this.props.location}</h3>
        <ul className="stats-wrap">
          {Object.keys(this.props.stats).map((stat, index) => (
            <li
              key={index}
              className={
                this.props.stats[stat] < 0.5 ? 'lowStats' : 'yearStats'
              }
            >
              {stat} : {this.props.stats[stat]}
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

DistrictCard.propTypes = {
  location: PropTypes.string,
  stats: PropTypes.object,
  retrieveCompare: PropTypes.func,
  comparedCards: PropTypes.array,
  removeCompareCard: PropTypes.func
};

export default DistrictCard;
