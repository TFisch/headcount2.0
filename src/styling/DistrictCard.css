.district-card {
  background-color: #5CDB95;
  border: 2px solid #05386B;
  font-family: 'Alegreya SC', serif;
  height: 18em;
  position:relative;
  -webkit-box-shadow:0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset;
     -moz-box-shadow:0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset;
          box-shadow:0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset;

  margin: .85em;
  padding-bottom: 1em;
  text-align: center;
  width: 15em;
}

.location-header {
  background-color:#05386B;
  color: #5CDB95;
  font-size: 1.1em;
  margin: 0;
}

.yearStats {
  color: #05386B;
  font-size: 1rem;
  margin: auto 0;
}

.lowStats {
  color: #747474;
  font-size: 1rem;
  font-weight: 600;
  margin: auto 0;
}

.stats-wrap {
  font-weight: 600; 
  padding: 1em;
}