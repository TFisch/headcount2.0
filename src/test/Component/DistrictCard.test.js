import React from 'react';
import { shallow, mount } from 'enzyme';
import DistrictCard from '../../Components/DistrictCard';

describe('DistrictCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <DistrictCard
        key={'key'}
        location={'location'}
        stats={'stats'}
        retrieveCompare={jest.fn()}
        comparedCards={[]}
        removeCompareCard={jest.fn()}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change selected state to true when clicked once', () => {
    const mockState = {
      selected: true
    };

    wrapper.simulate('click');
    expect(wrapper.state()).toEqual(mockState);
  });

  it('should change selected state back to false when clicked a second time', () => {
    const expectedState = {
      selected: true
    };

    const expectedChangeState = {
      selected: false
    };

    wrapper.simulate('click');
    expect(wrapper.state()).toEqual(expectedState);
    wrapper.simulate('click');
    expect(wrapper.state()).toEqual(expectedChangeState);
  });

  it('should have a list item with a class of lowStats if under .5', () => {
    const lowMockStats = { 2004: 0.3 };
    let wrapper = mount(
      <DistrictCard
        key={'key'}
        location={'Paris'}
        stats={lowMockStats}
        retrieveCompare={jest.fn()}
        comparedCards={[]}
        removeCompareCard={jest.fn()}
      />
    );

    expect(wrapper.find('.stats-wrap').children().hasClass('lowStats')).toEqual(true);

  });

  it('should have a list item with a class of yearStats if over .5', () => {
    const lowMockStats = { 2004: 0.6 };
    let wrapper = mount(
      <DistrictCard
        key={'key'}
        location={'Paris'}
        stats={lowMockStats}
        retrieveCompare={jest.fn()}
        comparedCards={[]}
        removeCompareCard={jest.fn()}
      />
    );

    expect(wrapper.find('.stats-wrap').children().hasClass('yearStats')).toEqual(true);

  });
});
