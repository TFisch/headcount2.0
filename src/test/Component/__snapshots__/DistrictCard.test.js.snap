// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictCard
    comparedCards={Array []}
    location="location"
    removeCompareCard={[Function]}
    retrieveCompare={[Function]}
    stats="stats"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="location-header"
>
          location
</h3>,
        <ul
          className="stats-wrap"
>
          <li
                    className="yearStats"
          >
                    0
                     : 
                    s
          </li>
          <li
                    className="yearStats"
          >
                    1
                     : 
                    t
          </li>
          <li
                    className="yearStats"
          >
                    2
                     : 
                    a
          </li>
          <li
                    className="yearStats"
          >
                    3
                     : 
                    t
          </li>
          <li
                    className="yearStats"
          >
                    4
                     : 
                    s
          </li>
</ul>,
      ],
      "className": "district-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "location",
          "className": "location-header",
        },
        "ref": null,
        "rendered": "location",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="yearStats"
>
              0
               : 
              s
</li>,
            <li
              className="yearStats"
>
              1
               : 
              t
</li>,
            <li
              className="yearStats"
>
              2
               : 
              a
</li>,
            <li
              className="yearStats"
>
              3
               : 
              t
</li>,
            <li
              className="yearStats"
>
              4
               : 
              s
</li>,
          ],
          "className": "stats-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "0",
                " : ",
                "s",
              ],
              "className": "yearStats",
            },
            "ref": null,
            "rendered": Array [
              "0",
              " : ",
              "s",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "1",
                " : ",
                "t",
              ],
              "className": "yearStats",
            },
            "ref": null,
            "rendered": Array [
              "1",
              " : ",
              "t",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2",
                " : ",
                "a",
              ],
              "className": "yearStats",
            },
            "ref": null,
            "rendered": Array [
              "2",
              " : ",
              "a",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "3",
                " : ",
                "t",
              ],
              "className": "yearStats",
            },
            "ref": null,
            "rendered": Array [
              "3",
              " : ",
              "t",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "4",
                " : ",
                "s",
              ],
              "className": "yearStats",
            },
            "ref": null,
            "rendered": Array [
              "4",
              " : ",
              "s",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="location-header"
>
            location
</h3>,
          <ul
            className="stats-wrap"
>
            <li
                        className="yearStats"
            >
                        0
                         : 
                        s
            </li>
            <li
                        className="yearStats"
            >
                        1
                         : 
                        t
            </li>
            <li
                        className="yearStats"
            >
                        2
                         : 
                        a
            </li>
            <li
                        className="yearStats"
            >
                        3
                         : 
                        t
            </li>
            <li
                        className="yearStats"
            >
                        4
                         : 
                        s
            </li>
</ul>,
        ],
        "className": "district-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "location",
            "className": "location-header",
          },
          "ref": null,
          "rendered": "location",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="yearStats"
>
                0
                 : 
                s
</li>,
              <li
                className="yearStats"
>
                1
                 : 
                t
</li>,
              <li
                className="yearStats"
>
                2
                 : 
                a
</li>,
              <li
                className="yearStats"
>
                3
                 : 
                t
</li>,
              <li
                className="yearStats"
>
                4
                 : 
                s
</li>,
            ],
            "className": "stats-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "0",
                  " : ",
                  "s",
                ],
                "className": "yearStats",
              },
              "ref": null,
              "rendered": Array [
                "0",
                " : ",
                "s",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "1",
                  " : ",
                  "t",
                ],
                "className": "yearStats",
              },
              "ref": null,
              "rendered": Array [
                "1",
                " : ",
                "t",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2",
                  " : ",
                  "a",
                ],
                "className": "yearStats",
              },
              "ref": null,
              "rendered": Array [
                "2",
                " : ",
                "a",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "3",
                  " : ",
                  "t",
                ],
                "className": "yearStats",
              },
              "ref": null,
              "rendered": Array [
                "3",
                " : ",
                "t",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "4",
                  " : ",
                  "s",
                ],
                "className": "yearStats",
              },
              "ref": null,
              "rendered": Array [
                "4",
                " : ",
                "s",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
